src/FuncSharp/Option/IOptionExtensions.cs

```diff
@@ -313,5 +314,61 @@ public static bool Is<A>(this IOption<A> option, Func<A, bool> predicate)
 
             return Try.Error<A, E>(e(Unit.Value));
         }
+
+        /// <summary>
+        /// Maps value of the current <see cref="IOption{A}"/> (if present) into a new value using the specified function and
+        /// returns a new <see cref="IOption{A}"/> (with that new value) wrapped in a <see cref="System.Threading.Tasks.Task"/>.
+        /// </summary>
+        public static async Task<IOption<B>> MapAsync<A, B>(this IOption<A> option, Func<A, Task<B>> f)
+        {
+            if (option.NonEmpty)
+            {
+                return Option.Valued(await f(option.GetOrDefault()));
```

Comment:
since the `option` instance holds a value, you can safely call `.Get()` method, right?

Reply:
Yes, but actually `Get` has extra logic in it - GetOrDefault just gives the value without any conditions :D

Reply:
I started thinking to introduce an internal property `Value`